google3/third_party/tensorflow_model_optimization/opensource/OFFICIAL_MODELS.md

```diff
@@ -0,0 +1,67 @@
+# Production Example and Convergence Testing
+
+One of the contribution requirements of a new technique is a training
+example for using the technique on realistic tasks and models. At the same
+time, the example will come with convergence tests to ensure that users can
+continue to achieve the benefits described in the documentation.
+
+The target task and model will be one of the ones available
+in the [TensorFlow Official
+Models](https://github.com/tensorflow/models/tree/master/official),
+excluding trivial models and tasks such as MNIST.
+
+This example and test will either live in TensorFlow Official Models
+or in a fork of it, with the following process.
+
+1. The Official Models team will review the proposal RFC and determine whether the
+   example and tests for the technique will live in TF Official Models or a fork,
+   based on the motivation for a user to use it and compatibility requirements.
```

Comment:
Can you share examples of the criteria to choose between the Official Models or a fork? Why is the fork needed at all?

Reply:
Yes I'll add something, based on comments from Official Models.

In general, I'd see it as a tradeoff between increasing usability and code complexity and maintenance versus the benefits to a general Official Models user, while also considering the degree of support (e.g. consider research-level vs production-level ownership). I know my comment is ambiguous (and this is something that's inherently ambiguous), and it's something that needs to be worked out. 

From personally considering their perspective, note that there is a plethora of features (tf.distribute, mixed precision, model coverage (where it's important to maintain the reference implementation while also considering the people want to modify the architecture for different purposes), tf.function vs eager) that they need to consider in an incremental and stable manner, given that many users are using the models for reference.

This is the reason why it's important to modularize the model optimization code (and let's say, only show all the model optimization flags if it's a user interested in model optimization), which is what I'm doing.